src/main/java/com/iexec/sms/secret/compute/TeeTaskComputeSecretService.java

```diff
@@ -76,13 +77,29 @@ public boolean isSecretPresent(OnChainObjectType onChainObjectType,
                                    SecretOwnerRole secretOwnerRole,
                                    String secretOwner,
                                    String secretKey) {
-        return getSecret(
+
+        final TeeTaskComputeSecretHeader key = new TeeTaskComputeSecretHeader(
+                onChainObjectType,
+                deployedObjectAddress,
+                secretOwnerRole,
+                secretOwner,
+                secretKey
+        );
+        final Boolean found = lookSecretExistenceInCache(key);
+        if (found != null) {
+            return found;
+        }
+
+        final boolean isPresentInDB = getSecret(
```

Comment:
In a future PR, we may change the signature of `getSecret` to get a header as parameter, instead of a list of objects. This would avoid to build twice the same header.